app/flatpak-builtins-config.c

```diff
@@ -64,49 +68,73 @@ looks_like_a_language (const char *s)
 }
 
 static gboolean
-looks_like_a_region (const char *s)
+looks_like_a_territory (const char *s)
 {
-  g_auto(GStrv) locale = g_strsplit (s, "_", 3);
-  int i;
-  int len;
+  gsize len = strlen (s);
+  gsize i;
 
-  if (!looks_like_a_language(locale[0]))
+  if (len < 2)
     return FALSE;
 
-  if (locale[1] != NULL)
+  for (i = 0; i < len; i++)
     {
-      len = strlen (locale[1]);
-
-      // This can be either GB or Latn
-      if (len < 2 || len > 4)
+      if (!g_ascii_isalpha (s[i]) || !g_ascii_isupper (s[i]))
         return FALSE;
-
-      for (i = 0; i < len; i++)
-        {
-          if ((locale[2] == NULL) && (!g_ascii_isalpha (locale[1][i]) || !g_ascii_isupper (locale[1][i])))
-            return FALSE;
-          else if (!g_ascii_isalpha (locale[1][i]))
-            return FALSE;
-        }
     }
 
-  if (g_strv_length (locale) > 2)
-    {
-      len = strlen (locale[2]);
+  return TRUE;
+}
 
-      if (len < 2 || len > 3)
-        return FALSE;
+static gboolean
+looks_like_a_codeset_or_modifier (const char *s)
+{
+  gsize len = strlen (s);
+  gsize i;
 
-      for (i = 0; i < len; i++)
-        {
-          if (!g_ascii_isalpha (locale[2][i]) || !g_ascii_isupper (locale[2][i]))
-            return FALSE;
-        }
+  if (len < 1)
+    return FALSE;
+
+  for (i = 0; i < len; i++)
+    {
+      if (!g_ascii_isalnum (s[i]) && s[i] != '-')
+        return FALSE;
     }
 
   return TRUE;
 }
 
+static gboolean
+looks_like_a_locale (const char *s)
```

Comment:
Should we accept codesets to be inserted in these configurations? Because this is mainly related to content, and I don't think it really matters to have this specified. Or is it just a matter of going by the book with the solution intended?

Reply:
As above, I don’t think it hurts, but probably doesn’t make sense as an explicit use case.